### PR TITLE
Add `preserveConsecutiveUppercase` option

### PR DESCRIPTION
**Pull request by DanielSeehausen**
This PR builds on @mogafk's #61
closes #61 

#### Per discussion in #61:
- [adds additional tests](https://github.com/sindresorhus/camelcase/pull/61#discussion_r389480686)
- updates inline documentation/examples
  - [documentation](https://github.com/sindresorhus/camelcase/pull/61#discussion_r389480168)
  - [examples](https://github.com/sindresorhus/camelcase/pull/61#discussion_r389480244)
- [updates naming convention](https://github.com/sindresorhus/camelcase/pull/61#discussion_r389481179)

#### Additional Changes
- test coverage cases present in only some option permutations now in all
- updates #61 to handle unicode characters
- abstracts certain minor functions now that there are multiple options (pascal case, preserveConsecutiveUppercase), being handled for read/maintain-ability (postProcess now explicitly called `handlePascalCase`)
- [updates default options arg](https://github.com/sindresorhus/camelcase/compare/master...DanielSeehausen:save-multiple-capitalize?expand=1#diff-168726dbe96b3ce427e7fedce31bb0bcR46)

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -38,8 +48,11 @@ camelCase('Foo-Bar', {pascalCase: true});
 camelCase('--foo.bar', {pascalCase: false});
 //=> 'fooBar'
 
-camelCase('foo bar');
-//=> 'fooBar'
```
Comment (DanielSeehausen) [inline index.d.ts:42]: this got moved up a few lines

Comment (sindresorhus) [reply index.d.ts:42]: You need to update the readme too.

**Comment by sindresorhus** [comment]
Can you make sure there's a test for https://github.com/sindresorhus/camelcase/pull/64/commits/b3ce30d75555778e0036dbd53ad005c9734311ee?

File: readme.md [hunk 2]
```diff
@@ -71,6 +71,13 @@ Default: `false`
 
 Uppercase the first character: `foo-bar` → `FooBar`
 
+##### preserveConsecutiveUppercase
+
+Type: `boolean`\
+Default: `false`
+
+Save sequence capital chars: `foo-BAR` → `fooBAR`
```
Comment (sindresorhus) [inline readme.md:79]: The readme should be in sync with index.d.ts

**Comment by sindresorhus** [comment]
@DanielSeehausen Bump :)

**Comment by sindresorhus** [comment]
Closing for lack of activity. https://github.com/sindresorhus/camelcase/issues/76

**Pull request closed without merging**